src/MomentoTaggedCache.php

```diff
@@ -49,9 +48,30 @@ public function get($key, $default = null): mixed
     /**
      * @throws UnknownError
      */
-    public function flush()
+    public function flush(): bool
     {
-        throw new UnknownError("flush operations is currently not supported.");
+        $tags = $this->tags->getNames();
+        if (!self::validateTags($tags)) {
+            throw new UnknownError("Empty tag is not valid. Please provide non-empty tags to flush.");
+        }
+        foreach ($tags as $tag) {
+            $keys = $this->store->setFetch($tag);
+            if (!$keys) {
+                return false;
+            } else {
+                foreach ($keys as $key) {
+                    $forgetResult = $this->store->forget($key);
+                    if (!$forgetResult) {
+                        return false;
+                    }
```

Comment:
will forget return false if the thing it tried to forget wasn't there? Is returning false and leaving a partial state behind the right thing to do here?